src/test/java/io/confluent/connect/elasticsearch/ElasticsearchSinkTaskTest.java

```diff
@@ -72,4 +79,39 @@ public void testPutAndFlush() throws Exception {
     verifySearchResults(records, true, false);
   }
 
+  @Test
+  public void testPutWithTopicInCaps() {
+    // We should as well test that writing a record with a previously un seen record will create
+    // an index following the required elasticsearch requirements of lowercasing.
+    InternalTestCluster cluster = ESIntegTestCase.internalCluster();
+    cluster.ensureAtLeastNumDataNodes(3);
+    Map<String, String> props = createProps();
+
+    ElasticsearchSinkTask task = new ElasticsearchSinkTask();
+
+    String key = "key";
+    Schema schema = createSchema();
+    Struct record = createRecord(schema);
+
+    SinkRecord sinkRecord = new SinkRecord(TOPIC_IN_CAPS,
+            PARTITION_113,
+            Schema.STRING_SCHEMA,
+            key,
+            schema,
+            record,
+            0 );
+
+    try {
+      task.start(props, client);
+      task.open(new HashSet<>(Collections.singletonList(TOPIC_IN_CAPS_PARTITION)));
+      task.put(Collections.singleton(sinkRecord));
+      assertTrue("A topic name not in lowercase was created in Elasticsearch", true);
```

Comment:
This assertion doesn't really check anything, since you're always passing `true` into the method. Is this intentional?

Don't we really want to check whether the index was created in the ES cluster? Without that, how are we sure that this functionality works?

Reply:
If you try to create an index with caps an exception will be throw, should look something like:

```java
org.apache.kafka.connect.errors.ConnectException: Could not create index 'AnotherTopicInCaps': {"root_cause":[{"type":"invalid_index_name_exception","reason":"Invalid index name [AnotherTopicInCaps], must be lowercase","index":"AnotherTopicInCaps"}],"type":"invalid_index_name_exception","reason":"Invalid index name [AnotherTopicInCaps], must be lowercase","index":"AnotherTopicInCaps"}
	at __randomizedtesting.SeedInfo.seed([A1BAC01989C87C:D2B7A48645E95AFE]:0)
	at io.confluent.connect.elasticsearch.ElasticsearchWriter.createIndicesForTopics(ElasticsearchWriter.java:268)
	at io.confluent.connect.elasticsearch.ElasticsearchSinkTask.open(ElasticsearchSinkTask.java:120)
	at io.confluent.connect.elasticsearch.ElasticsearchSinkTaskTest.testPutWithTopicInCaps(ElasticsearchSinkTaskTest.java:106)
```
I've learn this pattern over the year, probably contaminated with ruby and other languages, so yes to answer your question. I now the assertTrue does nothing, is mostly to show the message, the real check for this to work in on the fail part.

I am happy to write it more politely and learn, please let me know how it should look? should I just remove the assert?

Reply:
Very useful feedback @rhauch, pushing again my java testing skills 😄 forward, love that!